napari/layers/shapes/shapes.py

```diff
@@ -655,6 +655,9 @@ def data(self, data):
             )
 
         self._data_view = ShapeList(ndisplay=self._slice_input.ndisplay)
+        self._data_view.slice_key = np.array(self._slice_indices)[
```

Comment:
Fix works for me. But I wonder if it should be applied to other cases where we create an instance of `ShapeList`? I was thinking about passing more to `ShapeList` itself (so this happens every time), but that is more complex.

Reply:
I think Yes. In the past, I meet a similar error, that I fix in a similar way. But I'm not using the Shapes layer, so I'm not deep diving to understand all motivations.